### ISSUE TITLE
what the training data's format？

### ISSUE THREAD

**Issue by Haitons** [Websail-NU/torch-defseq#3]
I don‘t konw what the training data’s format and what does input and output look like?How to compute the loss？The data and target make me confused.Can you answer for me？

**Comment by northanapon** [comment]
The format of the training data is

```
<word being defined>\t<pos>\t<definitions>\n
...
```

We didn't actually use `<pos>`, so you can add anything there as long as it is not a tab or a newline. You can see our [valid.txt](https://github.com/Websail-NU/torch-defseq/blob/master/data/commondefs/valid.txt) for an example.

**Comment by Haitons** [comment]
For example:The traing data is `implosion <s> a sudden inward collapse</s>`and the target is ` <s> a sudden inward collapse</s>`.Is it right?

**Comment by northanapon** [comment]
The training data that the model uses is in a format as you mentioned:
```<s> implosion <def> a sudden inward collapse </s>```
It might be slightly different in the detail.

We are using a language modeling objective. Given previous words, the model is to predict the probability distribution of the next word. For each time step, the model reads the word being defined, the current word, and the previous state as an input and predicts a probability distribution as an output. The actual word in the definition is then the target. For example, when the model is going to predict ```sudden```, the input is an embedding of ```implosion```, an embedding of ```a```, and a previous state (LSTM). The target for the output distribution is ```sudden```.

**Comment by Haitons** [comment]
Thank you.I konw that.And when I generate the definition,How to compute the BLEU?I find two file in`eval`:`bleu.py`and`compute_bleu.py`.What is the difference between them?I konw  the`comput_bleu.py` use nltk.corpus_bleu to compute the BLEU.But I got bad result...I am so confused....

**Comment by northanapon** [comment]
`bleu.py` uses `sentence_bleu` from Moses project. BLEU is designed to evaluate MT systems. When comparing semantics (with only a few referrece texts), we usually use one of a [smoothing method](http://acl2014.org/acl2014/W14-33/pdf/W14-3346.pdf). Though I am not sure what exact method that the Moses' `sentence_bleu` uses.

**Comment by Haitons** [comment]
OK.So the result in the paper computed by the `bleu.py`?And what the blue_path`./sentence-bleu`?

**Comment by northanapon** [comment]
That's a compiled file from Moses' [sentence-bleu.cpp](https://github.com/moses-smt/mosesdecoder/blob/master/mert/sentence-bleu.cpp)

**Comment by Haitons** [comment]
Thank you very much!Is the BLEU of the paper in Table 5 calculated from `bleu.py`?

**Comment by northanapon** [comment]
Yes!
